### DISCUSSION TITLE
Edit one column in database with two fields in form

### DISCUSSION THREAD

**Discussion by sebastianpopp** [filamentphp/filament#9344] [Help] [answered]
Hi there,

I have a `point` column in my mysql database to hold geocoordinates (using [matanyadaev/laravel-eloquent-spatial](https://github.com/MatanYadaev/laravel-eloquent-spatial) for casting). I would like to be able to edit that field in my form using two `TextInput` fields. Populating the fields already works, but I don't understand how I can save those two fields back into the database. Here's my code for those two fields:

```php
Forms\Components\TextInput::make('latitude')
    ->numeric()
    ->minValue(-90)
    ->maxValue(90)
    ->step(0.0000000001)
    ->afterStateHydrated(function (Forms\Components\TextInput $component, Model $record) {
        $component->state($record->location->latitude);
    })
    ->dehydrateStateUsing(function ($state, Model $record) {
        $record->location->latitude = $state;
    })
    ->dehydrated(false),
Forms\Components\TextInput::make('longitude')
    ->numeric()
    ->minValue(-180)
    ->maxValue(180)
    ->step(0.0000000001)
    ->afterStateHydrated(function (Forms\Components\TextInput $component, Model $record) {
        $component->state($record->location->longitude);
    })
    ->dehydrateStateUsing(function ($state, Model $record) {
        $record->location->longitude = $state;
    })
    ->dehydrated(false),
```

Any suggestions would be much appreciated.

**Comment by leandrocfe** [comment, marked as answer]
Maybe customizing the data before creating/editing?

https://filamentphp.com/docs/3.x/panels/resources/creating-records/#customizing-data-before-saving

https://filamentphp.com/docs/3.x/panels/resources/editing-records/#customizing-data-before-saving

**Reply by sebastianpopp** [reply]
Thank you! That's what I was looking for

```php
protected function mutateFormDataBeforeSave(array $data): array
{
    $data['location'] = new Point($data['latitude'], $data['longitude']);

    return $data;
}
```
